src/main/java/com/streamr/broker/QueueProducer.java

```diff
@@ -19,7 +20,9 @@ public QueueProducer(BlockingQueue<StreamMessage> queue, Stats stats) {
 	public void accept(StreamMessage msg) {
 		try {
 			queue.put(msg);
-			stats.onReadFromKafka(msg);
+			for (Stats s: stats) {
```

Comment:
BTW this syntax also creates an `Iterator` object under the hood! 

This is definitely a micro-optimization, but if we want to save on garbage creation and method calls in this very hot method that gets called for every single message, thousands of times every second, we can use the classic syntax since we now use native arrays:

```
for (int i=0; i<stats.length; i++) {
    s[i].onReadFromKafka(msg);
}
```

Which only creates one `int` of garbage on each call, or taking it to ridiculous lengths (not necessary IMO):

```
int i; // reserve memory for the index variable elsewhere, only once

// ...

for (i=0; i<stats.length; i++) {
    s[i].onReadFromKafka(msg);
}
```